### PR TITLE
fix(AppMain): Made all banners global and stacked

### PR DESCRIPTION
**Pull request by igor-sirotin**
Fixes https://github.com/status-im/status-desktop/issues/6636

### Banners

Banners are now global and stacked:
1. Backup seed phrase [[Figma link]](https://www.figma.com/file/idUoxN7OIW2Jpp3PMJ1Rl8/%E2%9A%99%EF%B8%8F-Settings-%7C-Desktop?node-id=884%3A93419)
2. TestNet mode enabled [[Figma link]](https://www.figma.com/file/idUoxN7OIW2Jpp3PMJ1Rl8/%E2%9A%99%EF%B8%8F-Settings-%7C-Desktop?node-id=1930%3A136721)
3. New version available [[Figma link]](https://www.figma.com/file/17fc13UBFvInrLgNUKJJg5/Kuba%E2%8E%9CDesktop?node-id=305%3A65943)
4. Connection lost / restored (design not found) 

NOTE: Import from Discord [[Figma link]](https://www.figma.com/file/HOCnvlM62U8ASbtI5wwcgv/Status-Desktop---Import-Discord-Community-Flow?node-id=13%3A55603) - implemented [here](https://github.com/status-im/status-desktop/issues/6934), out of scope of this pr

All banners are divided into 2 types:

- signal-driven (Connection, Update) - always fixed to the top
- property-driven (TestNet, Backup) - appended below the last banner on signal

Stacked banners design: https://www.figma.com/file/17fc13UBFvInrLgNUKJJg5/Kuba%E2%8E%9CDesktop?node-id=15853%3A474834

### Testnet disable confirmation dialog
Implemented per [design](https://www.figma.com/file/idUoxN7OIW2Jpp3PMJ1Rl8/%E2%9A%99%EF%B8%8F-Settings-%7C-Desktop?node-id=1938%3A135119).

## Affected areas

AppMain

## Some logic behind this

"Import from Discord" is currently out of scope, since it's not merged to master yet.
I also noted that the "Update available" banner doesn't match designs a bit, but it's better to update it in a separate PR.

All banners are divided into 2 types:
- signal-driven (Connection, Update) - always fixed to the top
- property-driven (TestNet, Backup) - appended below the last banner on signal

## Screenshots

This one show the actual banners separation. Did this at late stage, don't want to recreate all the screenshots 🙂 
image

image
image
image
image

## Screencast

https://user-images.githubusercontent.com/25482501/188175779-4005e6c5-8ee3-4e79-9dd2-ff6901d271bf.mov

### CODE REVIEW

**Comment by caybro** [comment]
Is this still WIP? I see a lot of commented/temp code

File: ui/app/mainui/AppMain.qml [hunk 1]
```diff
@@ -683,6 +808,12 @@ Item {
                         }
                     }
                 }
+
+//                StatusBaseText {
+//                    anchors.centerIn: parent
+//                    font.pixelSize: 100
+//                    text: parent.height
+//                }
```
Comment (caybro) [inline ui/app/mainui/AppMain.qml:816]: A leftover?

File: ui/imports/shared/panels/ModuleWarning.qml [hunk 2]
```diff
@@ -3,111 +3,189 @@ import QtQuick.Controls 2.13
...
+        closeButtonMouseArea.clicked(null)
     }
 
-    Row {
-        spacing: Style.current.halfPadding
-        anchors.horizontalCenter: parent.horizontalCenter
-        anchors.verticalCenter: parent.verticalCenter
+    implicitHeight: root.active ? content.implicitHeight : 0
+    visible: implicitHeight > 0
+    clip: true
```
Comment (caybro) [inline ui/imports/shared/panels/ModuleWarning.qml:55]: Really need clipping?

File: ui/imports/shared/panels/ModuleWarning.qml [hunk 3]
```diff
@@ -3,111 +3,189 @@ import QtQuick.Controls 2.13
...
-            text: root.text
-            font.pixelSize: 13
-            anchors.verticalCenter: parent.verticalCenter
-            color: Style.current.white
+    onActiveChanged: {
+        active ? showAnimatinon.start() : hideAnimatinon.start()
+    }
+
+    NumberAnimation {
+        id: showAnimatinon
```
Comment (caybro) [inline ui/imports/shared/panels/ModuleWarning.qml:62]: typo

File: ui/imports/shared/panels/ModuleWarning.qml [hunk 4]
```diff
@@ -3,111 +3,189 @@ import QtQuick.Controls 2.13
...
-        Button {
-            width: btnWidth
-            height: 24
-            contentItem: Item {
-                anchors.fill: parent
-                Text {
-                    text: btnText
-                    font.pixelSize: 13
+    NumberAnimation {
+        id: hideAnimatinon
```
Comment (caybro) [inline ui/imports/shared/panels/ModuleWarning.qml:78]: typo

File: ui/imports/shared/panels/ModuleWarning.qml [hunk 5]
```diff
@@ -3,111 +3,189 @@ import QtQuick.Controls 2.13
...
+            }
+        }
+
+        color: baseColor
+
+        Behavior on color {
+            ColorAnimation {
+                duration: 150
+            }
+        }
```
Comment (caybro) [inline ui/imports/shared/panels/ModuleWarning.qml:122]: Need this? How realistic it is for the module to change type? :)

**Review by caybro — COMMENTED** [review]
Looks good, just some really minor stuff

File: ui/imports/shared/panels/ModuleWarning.qml (hunk 5 above)
Comment (caybro) [reply ui/imports/shared/panels/ModuleWarning.qml:121]: Ah right, that's what I was missing :)

**Review by caybro — APPROVED** [review]
Looks good from my POV

**Review by anastasiyaig — APPROVED** [review]

**Pull request merged**